addon/components/detail.js

```diff
@@ -509,7 +509,7 @@ export default Component.extend(PropTypeMixin, {
     registerComponentForFormValueChanges (component) {
       // Make sure when component is destroyed it unregisters for changes
       component.on('willDestroyElement', () => {
-        this.unregisterComponentForFormValueChanges(component)
+        component.unregisterForFormValueChanges(component)
```

Comment:
Wonder if it would be good to test if `unregisterForFormValueChanges` gets passed so consumers don't have integration tests breaking when they get this change.